java/server/src/org/openqa/selenium/grid/distributor/local/LocalDistributor.java

```
@@ -397,6 +399,18 @@ private LocalDistributor add(Node node, NodeStatus status) {
     return this;
   }
 
+  @Override
+  public void drain(UUID nodeId) {
+    Lock writeLock = lock.writeLock();
+    writeLock.lock();
+    try {
+      hosts.stream().filter(node -> nodeId.equals(node.getId())).findFirst()
+          .orElseThrow().setHostStatus(Host.Status.DRAINING);
```

Throw something sensible rather than whatever `orElseThrow` throws by default